cmd/main.go

```diff
@@ -143,142 +175,165 @@ func main() {
+	return scope.ClientConfig{Token: linodeToken},
+		scope.ClientConfig{Token: linodeDNSToken, BaseUrl: linodeDNSURL, RootCertificatePath: linodeDNSCA}
+}
 
+// setupManager initializes and returns a new manager instance with the provided configurations.
+// It sets up the REST configuration, metrics server options, and registers controllers and webhooks.
+func setupManager(flags flagVars, linodeConfig, dnsConfig scope.ClientConfig) manager.Manager {
 	restConfig := ctrl.GetConfigOrDie()
-	restConfig.QPS = float32(restConfigQPS)
-	restConfig.Burst = restConfigBurst
+	restConfig.QPS = float32(flags.restConfigQPS)
+	restConfig.Burst = flags.restConfigBurst
 	restConfig.UserAgent = fmt.Sprintf("CAPL/%s", version.GetVersion())
 
+	var tlsOpts []func(*tls.Config)
+	if !flags.secureMetrics {
+		tlsOpts = append(tlsOpts, func(c *tls.Config) {
+			setupLog.Info("disabling http/2")
+			c.NextProtos = []string{"http/1.2"}
+		})
+	}
+
+	metricsServerOptions := metricsserver.Options{
+		BindAddress:   flags.metricsAddr,
+		SecureServing: flags.secureMetrics,
+		TLSOpts:       tlsOpts,
+	}
+	if flags.secureMetrics {
+		metricsServerOptions.FilterProvider = filters.WithAuthenticationAndAuthorization
+	}
+
 	mgr, err := ctrl.NewManager(restConfig, ctrl.Options{
 		Scheme:                 scheme,
-		Metrics:                metricsserver.Options{BindAddress: metricsAddr},
-		HealthProbeBindAddress: probeAddr,
-		LeaderElection:         enableLeaderElection,
+		Metrics:                metricsServerOptions,
+		HealthProbeBindAddress: flags.probeAddr,
+		LeaderElection:         flags.enableLeaderElection,
 		LeaderElectionID:       "3cfd31c3.cluster.x-k8s.io",
-		// LeaderElectionReleaseOnCancel defines if the leader should step down voluntarily
-		// when the Manager ends. This requires the binary to immediately end when the
-		// Manager is stopped, otherwise, this setting is unsafe. Setting this significantly
-		// speeds up voluntary leader transitions as the new leader don't have to wait
-		// LeaseDuration time first.
-		//
-		// In the default scaffold provided, the program ends immediately after
-		// the manager stops, so would be fine to enable this option. However,
-		// if you are doing or is intended to do any operation such as perform cleanups
-		// after the manager stops then its usage might be unsafe.
-		// LeaderElectionReleaseOnCancel: true,
 	})
-	if mgr == nil || err != nil {
-		setupLog.Error(err, "unable to start manager")
+	if err != nil {
+		setupLog.Error(err, "unable to create manager")
+		os.Exit(1)
+	}
+
+	// Ensure mgr is not nil before proceeding
+	if mgr == nil {
+		setupLog.Error(errors.New("manager creation failed"), "manager is nil")
 		os.Exit(1)
 	}
 
-	if err = (&controller.LinodeClusterReconciler{
+	setupControllers(mgr, flags, linodeConfig, dnsConfig)
+
+	// Setup webhooks if enabled
+	if os.Getenv("ENABLE_WEBHOOKS") != "false" {
```

Comment:
probably want to use strconv.ParseBool here

Reply:
Updated it!